### DISCUSSION TITLE
Emacs 28 and project.el issue - Wrong type argument during switch project

### DISCUSSION THREAD

**Discussion by ldeck** [magit/magit#4758] [Support] [answered]
Having updated to Emacs 28 I'm seeing the following error when starting emacs, switching to a project and selecting 'm' to open magit.

**Messages**
```text
Configuring forge-owned-accounts: (ldeck)
funcall-interactively: Wrong type argument: symbolp, (autoload "magit-extras" "Run `magit-status' in the current project's root." t nil)
```

What might be broken? Thanks!

Relevant hm-init.el file configuration for magit and project.el
```elisp
(use-package forge
  :after (magit)
  :config
  (setq forge-owned-accounts '())
  (let ((forgeAccounts
          (or
            (getenv "FORGE_OWNED_ACCOUNTS")
            (getenv "USER"))))
    (if forgeAccounts
      (let ((accountsList (list forgeAccounts)))
        (progn
          (message "Configuring forge-owned-accounts: %s" accountsList)
          (add-to-list 'forge-owned-accounts accountsList)
          ))))
  )

(use-package magit
  :bind (
          ("C-c g" . magit-status)
          )
  :commands (magit-project-status)
  :hook (magit-mode . (lambda ()
                        (interactive)
                        (transient-append-suffix 'magit-commit "a"
                          '("n" "Reshelve commit" magit-commit-reshelve))
                        (transient-append-suffix 'magit-rebase "s"
                          '("t" "Reshelve since" magit-reshelve-since))
                        (transient-append-suffix 'magit-push "-n"
                          '("=O" "Set extra push option #1" "--push-option=" read-from-minibuffer))
                        (transient-append-suffix 'magit-push "-n"
                          '("=P" "Set extra push option #2" "--push-option=" read-from-minibuffer))
                        (transient-append-suffix 'magit-push "-n"
                          '("-S" "Skip gitlab pipeline creation" "--push-option=ci.skip"))
                        ))

  :init
  (setq magit-git-executable (file-truename (locate-file "git" exec-path)))

  :config
  (add-to-list 'git-commit-style-convention-checks
    'overlong-summary-line)
  )

(use-package project
  :bind-keymap (
                 ("C-x p" . project-prefix-map)
                 )
  :bind (:map project-prefix-map
          ("m" . magit-project-status)
          )
  :commands (project-root)
  :config
  (add-to-list 'project-switch-commands '(magit-project-status "Magit") t)
  )

(use-package treemacs-magit
  :after (treemacs magit)
  :defer t)
```

**Comment by tarsius** [comment, marked as answer]
You could try deleting `magit-autoloads.el` and regenerating it. One way to do that would be to un- and re-install magit.

**Reply by ldeck** [reply]
Is there another way to do it?  
I've included above snippets of the output generated by my [emacs.nix](https://github.com/ldeck/nix-home/blob/85eca28036ec863740d90c8bb963302a7cf6fb2a/lib/defaults/emacs.nix) config, but I've not manually included magit.

I tried disabling the packages, then regenerating. Starting emacs had no magit.
Then re-enabling, but the same issue occurs, though perhaps because there was no diff with the original.
I'll try a few more things, but if there's some other way that'd be good to know.
Thanks @tarsius !

**Reply by ldeck** [reply]
@tarsius funnily enough, if I switch to a project and choose to go to a file, directory, or something else... they work. 
I can then switch to another project and select 'm' for magit happily from that point on.
So it's something perhaps lazy loaded that's not initially available when I first start emacs.
Any ideas?

**Reply by ldeck** [reply]
Weirdly, it fails the 1st time, but works the 2nd time, so I don't understand what's missing unless there's something lazily loaded that's not been loaded the first time.
